### PR TITLE
255 kan updates

### LINKED ISSUE #255: KAN updates
Introduce several improvements to the KAN implementation:
- Align the KAN constructor behavior with the MLP class for consistency.
- Add a seed parameter to ensure reproducibility during optimization.
- Store optimizer, schedulers, and dataloaders as class attributes during training.
- Refactor the scheduler definition in the fit method for improved clarity and readability.
- Align the `create_optimized_model()` method with the MLP class for consistency.
- Include additional minor improvements and cleanups.

### PR DESCRIPTION
**Pull request by fermingut**
Closes #255

### CODE REVIEW

**Comment by fermingut** [comment]
@ArnauMiro I don't know if we should merge this before of after #249

**Comment by ArnauMiro** [comment]
@fermingut no, #249 has been fully reviewed and has preference now, otherwise we will never merge it.

File: pyLOM/NN/architectures/kan.py [hunk 1]
```diff
@@ -7,10 +7,12 @@
 import torch.optim as optim
 from torch.utils.data import DataLoader
 
+from pyLOM.NN.utils import set_seed
```
Comment (ArnauMiro) [inline pyLOM/NN/architectures/kan.py]: use a relative import

File: pyLOM/NN/architectures/kan.py [hunk 2]
```diff
@@ -7,10 +7,12 @@
...
+
 from ... import cr, pprint  # pyLOM/__init__.py
```
Comment (ArnauMiro) [inline pyLOM/NN/architectures/kan.py]: Organize pyLOM imports

File: pyLOM/NN/architectures/kan.py [hunk 3]
```diff
@@ -254,23 +274,27 @@ def closure():
                     torch.stack([p.grad.norm() for p in self.parameters() if p.grad is not None])
                 )
                 grad_norms.append(total_norm.item())
-                if scheduler_type != "ReduceLROnPlateau":
-                    self.scheduler.step()
-                    current_lr = self.optimizer.param_groups[0]["lr"]
-                    current_lr_vec.append(current_lr)
+                # if scheduler_type != "ReduceLROnPlateau":
```
Comment (ArnauMiro) [inline pyLOM/NN/architectures/kan.py]: Commented code not useful?

**Review by ArnauMiro — CHANGES_REQUESTED** [review]

File: pyLOM/NN/architectures/kan.py (hunk 3 above)
Comment (fermingut) [reply pyLOM/NN/architectures/kan.py]: We were working on it, it is fixed now

**Review by ArnauMiro — APPROVED** [review]

**Comment by ArnauMiro** [comment]
@fermingut we can merge anytime you want.
